ocpmodels/trainers/base_trainer.py

```diff
@@ -527,15 +522,16 @@ def _compute_loss(self, out, batch):
         if "grad_input" in self.config["task"]:
             if self.config["dataset"].get("normalize_labels", True):
                 grad_target_normed = self.normalizers["grad_target"].norm(
-                    batch.force
+                    force_target
                 )
             else:
-                grad_target_normed = batch.force
+                grad_target_normed = force_target
 
             # Force coefficient = 30 has been working well for us.
             force_mult = self.config["optim"].get("force_coefficient", 30)
             if self.config["task"].get("train_on_free_atoms", False):
-                mask = batch.fixed == 0
+                fixed = torch.cat([batch.fixed for batch in batch_list])
+                mask = fixed == 0
                 loss.append(
                     force_mult
                     * self.criterion(
```

Comment:
We could consider redefining `_forward` and `_compute_loss` in `ForcesTrainer` instead. That way these changes won't break backward compatibility to other trainers.

Reply:
Yeah this makes a lot more sense and will break a lot less things. I'll move all the new data parallel bits to the forces trainer instead.